apimodels/agent_models.go

```diff
@@ -114,8 +125,38 @@ type EbsDevice struct {
 	SnapshotID string `mapstructure:"ebs_snapshot_id" json:"ebs_snapshot_id"`
 }
 
-func (ch *CreateHost) Validate() error {
+type RegistrySettings struct {
+	Name     string `mapstructure:"registry_name" json:"registry_name"`
+	Username string `mapstructure:"registry_username" json:"registry_username"`
+	Password string `mapstructure:"registry_password" json:"registry_password"`
+}
+
+func (ch *CreateHost) ValidateDocker() error {
+	catcher := grip.NewBasicCatcher()
+	if ch.Distro == "" {
+		catcher.Add(errors.New("distro must be set"))
+	}
+	if ch.Image == "" || ch.Command == "" {
+		catcher.Add(errors.New("docker image and command must both be set"))
+	}
+
+	if ch.PollFrequency == 0 {
```

Comment:
Maybe `if ch.PollFrequency <= 0`, to fix if someone sets a negative poll frequency. Also, should we set some large max, like maybe an hour? Finally, what happens if someone sets a large-ish polling interval and the container exits? Do we not find out that the container has exited until after we have polled? If that's the case, it might make sense to have a smaller max than an hour, maybe not longer than a minute, or else tasks will stick around waiting for the containers to finish running for a long time.

Reply:
A timeout definitely seems reasonable but one minute might be too restrictive since the users are just polling for logs and might be running something long. Maybe let's settle on 5 minutes for now just to provide flexibility.